Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift

```diff
@@ -14,13 +14,18 @@ import SwiftSyntax
 
 // MARK: - HasCodeBlock
 
+// TODO: Consider removing `HasTrailingCodeBlock` and friends
+// It'd be much nicer if we didn't have any builder APIs taking strings. We
+// should just make convenvience inits where needed such that we could write:
+//   `IfStmt("if foo")` -> `IfStmt(conditions: [.expression("foo")])`
```

Comment:
This is what we had before I introduced `HasTrailingCodeBlock` and it was super verbose and I didn't like it. If you want, you can take a look at old versions of how classes or Strucks were being constructed in CodeGeneration.

Reply:
I'll take a look, but if it's too verbose we should fix that. `IfStmt("if foo")` seems really weird to me, we've said it's an `if` so why are we writing `if` again 😅? In general I'd just really like to steer away from methods that take a `String` and parse it, without it being obvious that's what they're doing.

Reply:
I assume you're referring to https://github.com/apple/swift-syntax/commit/2c04f3416a098b55d76080f3b977e0122fdb49f7? To take some examples from there -

Old:
```
SwitchCase(label: SwitchCaseLabel(caseItems: [CaseItem(pattern: ExpressionPattern(expression: MemberAccessExpr(name: "contextualKeyword")))]))
```
New:
```
SwitchCase("case .contextualKeyword:")
```
Suggested:
```
SwitchCase(pattern: ".contextualKeyword") { }

Old:
```
ExtensionDecl(
       leadingTrivia: node.documentation.isEmpty
         ? []
         : .docLineComment("/// \(node.documentation)") + .newline,
       extendedType: Type(node.type.shorthandName),
       inheritanceClause: TypeInheritanceClause { InheritedType(typeName: Type("ExpressibleByArrayLiteral")) }
     )
```
New:
```
ExtensionDecl("\(docComment)extension \(node.type.shorthandName): ExpressibleByArrayLiteral")
```
Suggested:
```
ExtensionDecl(
  leadingTrivia: docComment,
  extendedType: "\(node.type.shorthandName)",
  inheritances: ["ExpressibleByArrayLiteral"])
```

Reply:
I think the function declarations were one of the worst in the old API but maybe you're right that we can switch if every child node of these is passable from a string literal. Your examples definitely look good to me.